### PR TITLE
Add session attribution analysis with state-dep edges

### PR DESCRIPTION
**Pull request by ojongerius**
## What

Implements session attribution analysis (§4) to compute cross-agent state dependencies via shared file resources. Adds:

- `SessionAttribution()` reader method that extracts file identity from receipt JSON and builds a provable dependency graph
- `AttributionResult` struct with coverage metrics, per-node risk profiles, state-dependency edges, and blast-radius maps
- `/api/sessions/{sessionID}/attribution` HTTP endpoint
- Session graph visualization enhancements: state-dep edges (solid blue), delegation edges (dashed gray), risk rings, and interactive blast-radius display
- Comprehensive test coverage for edge cases (empty sessions, bidirectional collapse, multi-resource aggregation, session isolation)

## Why

Session attribution enables forensic analysis of cross-agent influence within a session. The state-dependency graph (built from `action.target.resource` paths) provides a provable signal of which agents affected which files, surfacing potential supply-chain risks. The coverage fraction warns when identity indexing is incomplete (shell/MCP receipts lack file paths). Move-op detection flags when path strings may not reliably identify file versions across renames.

## Fixes

- **`⚠ mv ops` false positives on deletes** — `has_move_ops` detection now requires exact match against `filesystem.file.move` or `filesystem.file.rename` instead of substring match (which incorrectly matched `filesystem.file.remove`)
- **Bidirectional state-dep edges** — when two agents alternately touched the same file, both A→B and B→A edges were emitted. Edge keys are now canonicalised (sorted) so any agent pair produces at most one undirected edge per shared resource
- **Empty-session handling** — sessions with no receipts now return 200 with empty coverage instead of error

## Checklist

- [x] `go test ./...` passes
- [x] `go vet ./...` passes
- [x] No real keys or secrets in the diff
- [x] New functionality includes tests (11 new test functions in reader_test.go, 2 in server_test.go)
- [x] Commit messages follow Conventional Commits
- [x] AGENTS.md not applicable (no project structure changes)

### CODE REVIEW

File: internal/store/reader.go [hunk 1]
```diff
@@ -880,6 +880,337 @@ func (r *Reader) ServerStats(since *string) ([]ServerStat, error) {
...
+				continue // same-agent re-touch; captured in blast radius
+			}
+			a, b := prev, next
+			if a > b {
+				a, b = b, a
+			}
+			ek := edgeKey{a, b}
+			if edgeResources[ek] == nil {
+				edgeResources[ek] = map[string]bool{}
+				edgeFrom[ek] = prev // first encounter determines causal direction
```
Comment (ojongerius) [inline internal/store/reader.go]: **Bug: `edgeFrom` direction is non-deterministic when two resources share the same agent pair with opposite temporal orderings.**

`fileIndex` is a `map[string][]touch`, so `range fileIndex` iterates resources in randomised order each run. `edgeFrom[ek]` is set on the *first resource encountered* for a given `edgeKey`. If agent B touched R1 before A, and agent A touched R2 before B, both produce `edgeKey{agentA,agentB}` — whichever resource the map iterator visits first wins, making `from_agent`/`to_agent` flip between requests for the same data.

Fix: elect the globally-earliest temporal first-mover during the loop, rather than relying on first encounter:

```go
// replace the nil-guard with a timestamp comparison
if cur, ok := edgeFrom[ek]; !ok || prev < cur {
    edgeFrom[ek] = prev
}
```

(ISO-8601 timestamps sort lexicographically, consistent with the `ORDER BY timestamp ASC` in the query.)

File: internal/server/static/index.html [hunk 2]
```diff
@@ -3571,9 +3585,11 @@ <h2 class="modal-title" id="forensic-modal-title">Forensic decryption key</h2>
...
 
     // buildSessionGraph extracts nodes (one per unique agent) and edges (delegation
-    // links) from a flat receipt array. The root/orchestrator agent has agentId null
-    // in the receipt data and is represented as '__root__' internally.
-    function buildSessionGraph(receipts) {
+    // links + cross-agent state deps) from a flat receipt array. The root/orchestrator
+    // agent has agentId null in the receipt data and is represented as '__root__'.
+    // The optional attribution param (from /api/sessions/{id}/attribution) enriches
+    // nodes with risk metadata and adds provable state-dependency edges.
+    function buildSessionGraph(receipts, attribution) {
```
Comment (ojongerius) [inline internal/server/static/index.html:3603]: **Bug: `renderInlineSessionGraph` calls `buildSessionGraph(receipts)` without the `attribution` argument (line ~3494), so the inline panel above the receipts table never shows state-dep edges, risk rings, or blast-radius data for any session.**

The modal path (line 3576) correctly fetches attribution and passes it here. The inline path doesn't, creating two inconsistent views of the same session.

Fix: fetch `/api/sessions/{id}/attribution` in `renderInlineSessionGraph` and pass the result as the second argument, the same way the modal does.

File: internal/server/static/index.html [hunk 3]
```diff
@@ -3632,37 +3672,63 @@ <h2 class="modal-title" id="forensic-modal-title">Forensic decryption key</h2>
...
+        const ringColor = n.maxRisk && riskRingColor[n.maxRisk];
+        // Risk ring rendered before the node circle so it sits behind it.
+        const riskRing = ringColor
+          ? `<circle cx="${p.x}" cy="${p.y}" r="${r + 5}" fill="none" stroke="${ringColor}" stroke-width="2.5" opacity="0.8"/>`
+          : '';
         return `
           <g class="sg-node" data-node-id="${safeId}" onclick="${clickFn}(this)"
              role="button" tabindex="0" aria-label="${escapeAttr(n.label)}${n.model ? ', ' + escapeAttr(n.model) : ''}, ${n.receiptCount} receipts"
              onkeydown="if(event.key==='Enter'||event.key===' '){event.preventDefault();${clickFn}(this)}">
+            ${riskRing}
```
Comment (ojongerius) [inline internal/server/static/index.html:3742]: **Bug (nearby at line 3737): SVG node sublabel unconditionally appends `…` even for agent IDs shorter than 12 characters.**

`sublabel` is set to `key.substring(0, 12)` (line ~3615). For keys shorter than 12 characters `substring` returns the full key — but the template at line 3737 still appends a literal `…`, falsely implying truncation. The CHANGELOG claims this was fixed, but `truncateHash` was only applied in `renderBlastRadius`, not in the SVG node label.

Fix:
```js
${n.sublabel ? `<text ...>${escapeHtml(truncateHash(n.sublabel, 12))}</text>` : ''}
```
or set `sublabel` via `truncateHash` at the assignment site so the source of truth is correct.

**Review by ojongerius — COMMENTED** [review]
Six findings from a high-effort review — two correctness bugs, one feature gap, one quadratic hot-path, and two simplifications.

File: internal/store/reader.go [hunk 4]
```diff
@@ -880,6 +880,337 @@ func (r *Reader) ServerStats(since *string) ([]ServerStat, error) {
...
+		}
+	}
+
+	for ak, nd := range nodeMap {
+		if len(nd.timestamps) == 0 {
+			continue
+		}
+		first, last := nd.timestamps[0], nd.timestamps[len(nd.timestamps)-1]
+		seen := map[string]bool{}
+		for _, rr := range all {
```
Comment (ojongerius) [inline internal/store/reader.go:1156]: **Efficiency: semantic-dep heuristic is O(agents × receipts) per HTTP request.**

The outer loop (line 1146) iterates every node; this inner loop rescans all rows for each one. For a 20-agent / 5 000-receipt session that's 100 000 iterations on every `GET /api/sessions/{id}/attribution`.

A single O(N) pre-pass can build `map[agentKey]set[resource]` for all agents at once; then the per-node count is just set arithmetic against the pre-built map, with no re-scan.

File: internal/store/reader.go [hunk 5]
```diff
@@ -880,6 +880,337 @@ func (r *Reader) ServerStats(since *string) ([]ServerStat, error) {
...
+
+// StatDepEdge is a provable state-dependency between two agents via a shared
+// resource: agent FromAgent touched resource R, then ToAgent touched the same R
+// (ordered by timestamp + chain + sequence). CrossAgent is always true here —
+// same-agent re-touches are captured in BlastRadius, not as drawn edges.
+type StatDepEdge struct {
+	FromAgent  string   `json:"from_agent"`
+	ToAgent    string   `json:"to_agent"`
+	Resources  []string `json:"resources"`
+	CrossAgent bool     `json:"cross_agent"`
```
Comment (ojongerius) [inline internal/store/reader.go:916]: **Simplification: `CrossAgent bool` is always `true` — the field is noise.**

Every emitted `StatDepEdge` sets `CrossAgent: true` (line 1103); the comment above even says so. The frontend's `if (sd.cross_agent)` checks (lines ~3645, ~3793–3794) are vacuous predicates that pass every edge unconditionally.

Consider removing the field from the struct and JSON response, and dropping the frontend guards. If a same-agent edge type is ever needed it can be a distinct type rather than a flag variant of this one.

File: internal/store/reader.go [hunk 6]
```diff
@@ -880,6 +880,337 @@ func (r *Reader) ServerStats(since *string) ([]ServerStat, error) {
...
+			ReceiptCount:  nd.receiptCount,
+			IdentityCount: nd.identityCount,
+			MaxRisk:       nd.maxRisk,
+			RiskProfile:   nd.riskProfile,
+			SemanticDeps:  nd.semCount,
+		})
+	}
+
+	totalReceipts := len(all)
+	identityReceipts := 0
```
Comment (ojongerius) [inline internal/store/reader.go:1194]: **Simplification: `identityReceipts` is a redundant third scan of `all`.**

The first loop (line 1033) already increments `nd.identityCount` for every row where `rr.resource != ""`. Summing over `nodeMap` gives the same total in O(agents) without the extra O(N) pass:

```go
for _, nd := range nodeMap {
    identityReceipts += nd.identityCount
}
```
